spec/controllers/work_groups_controller_spec.rb

```diff
@@ -1,119 +1,145 @@
-            start_time: DateTime.now + 1.day,
+            delivery_method: 'generate links',
+            start_time: Time.now + 60_000,
             start_date: Date.today
           }
         }
         expect(controller.instance_variable_get(:@work_group).turn_time).to eq(300_000)
         expect(WorkGroup.find_by_turn_time(300_000)).to be_instance_of(WorkGroup)
-        expect(controller.instance_variable_get(:@worksheet_urls)).to eq(['https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg'])
-        expect(response).to have_http_status(:success)
+        expect(controller.instance_variable_get(:@emails)).to eq('test1@gmail.com test2@gmail.com')
+        expect(controller.instance_variable_get(:@delivery_method)).to eq('generate links')
+        expect(controller.instance_variable_get(:@urls)).to eq('https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg')
       end
 
-      it 'assigns correct new work group given no params' do
-        worksheet1
+      it 'with no params' do
         get :new
-        # expect(controller.instance_variable_get(:@work_group).turn_time).to eq(nil)
-        expect(controller.instance_variable_get(:@worksheet_urls)).to eq(['https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg'])
-        expect(response).to have_http_status(:success)
+        expect(controller.instance_variable_get(:@work_group).turn_time).to eq(nil)
       end
     end
 
-    it 'Worksheet has wrong url and returns empty' do
-      # worksheet_template2
-      get :new
-      expect(controller.instance_variable_get(:@worksheet_urls)).to eq([])
-      expect(response).to have_http_status(:success)
+    context 'returns correct worksheet urls' do
+      it 'with correct url format' do
+        worksheet_template
+        get :new
+        expect(controller.instance_variable_get(:@worksheet_urls)).to eq(['https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg'])
+      end
+      it 'with incorrect url format' do
+        worksheet_template2
+        get :new
+        expect(controller.instance_variable_get(:@worksheet_urls)).to eq([])
+      end
     end
+  end
 
-  describe 'GET #create' do
+  describe '#create' do
     context 'success' do
-      it 'returns correct invitations and https success for existing students' do
-        student1
-        student2
-        classroom
-        put :create, params: {
+      before do
+        allow(WorkGroupDemoPrepper).to receive(:new).and_return(work_group_demo_prepper)
+        allow(work_group_demo_prepper).to receive(:call).and_return({ users: [student1, student2], work_group: work_group1 })
+        allow(DemoMailer).to receive(:with).and_return(demo_mailer)
+        allow(demo_mailer).to receive(:invite).and_return(demo_mailer)
+        allow(demo_mailer).to receive(:deliver_later).and_return(demo_mailer)
```

Comment:
could probably merge this 3 conditions into one using `.to receive_message_chain`  https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/message-chains